src/gilded_rose.js

```diff
@@ -1,62 +1,108 @@
-function Item(name, sell_in, quality) {
+'use strict'
+
+function Normal(name, sell_in, quality) {
   this.name = name;
   this.sell_in = sell_in;
   this.quality = quality;
 }
+Normal.prototype.update_quality = function () {
+  this.quality--
+  if (this.sell_in <= 0) {
+    changeQuality(this, -1)
+  }
+  this.quality = (this.quality < 0) ?  0 : this.quality
+  this.sell_in--
+};
+Normal.prototype.changeQuality = function (item, changeBy) {
+  item.quality = item.quality + changeBy
+  item.quality = (this.quality < 0) ? 0 : this.quality
+  item.quality = (this.quality > 50) ? 50 : this.quality
+};
 
-var items = []
+function AgedBrie(name, sell_in, quality) {
```

Comment:
The problem with using inheritance or new classes to represent each item is that the design is not about encapsulating `behavior` so the behavior is reusable. It works, sort-of, until you have to maintain the product over time. Now every new item you sell needs a new class (unless it's normal). 

Imagine another design where the behavior is encapsualted into funtions (or classes) - I create a new Item and give it the behaviors it needs. Let's use aging as an example. One difference between Aged Brie and a normal item is the quality behavior - so create an `IncreaseQuality` and `DescreaseQuality` aging behavior and attach them to the item. Now you call those functions to do the aging. When you add Wine,  there is no new code needed because you can reuse the `IncreaseQuality` behavior.